WAD2app/models.py

```diff
@@ -1,3 +1,130 @@
 from django.db import models
 
-# Create your models here.
+class Dog(models.Model):
+    #id = models.AutoField(primary_key=True)
+    name = models.CharField (max_length=20)
+
+    AGE_CHOICES = [
+        ('0','any age'),
+        ('1', '0 to 6 months'),
+        ('2', '7 to 12 months'),
+        ('3', '1 to 2 years'),
+        ('4', '2 to 5 years'),
+        ('5', '5 to 7 years'),
+        ('6', '8 years and over'),
+    ]
+    age = models.CharField (max_length=1, choices=AGE_CHOICES, default='0',)
+
+    SIZE_CHOICES = [
+        ('0', 'any size'),
+        ('S', 'small'),
+        ('M', 'medium'),
+        ('L', 'large'),
+        ('G', 'giant'),
+    ]
+    size = models.CharField (max_length=1, choices=SIZE_CHOICES, default='0',)
+
+    BREED_CHOICES = [
+        ('0', 'any breed'),
+        ('CB', 'a crossbreed'),
+        ('AK', 'akita'),
+        ('AL', 'alastian'),
+        ('AB', 'american bulldog'),
+        ('BE', 'beagle'),
+        ('BC', 'border collie'),
+        ('BO', 'boxer'),
+        ('BU', 'bulldog'),
+        ('CH', 'chihuahua'),
+        ('GE', 'german shepherd'),
+        ('GH', 'grey hound'),
+        ('1', 'other breeds'),
+    ]
+    breed = models.CharField (max_length=2, choices=BREED_CHOICES, default='0',)
+
+    description = models.TextField ()
+    #need to specify the parameters
+    image = models.ImageField (upload_to=None, height_field=None,
+                               width_field=None, max_length=100,)
+
+    preference1 = models.CharField (max_length=1)
+    preference2 = models.CharField (max_length=1)
+
+    STATUS_CHOICES = [
+        ('1', 'available'),
+        ('2', 'reserved'),
+        ]
+    status = models.CharField (max_length=1, default='1', choices=STATUS_CHOICES,)
+
+    def __str__(self):
+        return self.id
+
+class User (models.Model):
```

Comment:
```suggestion
class UserProfile(models.Model):
    favourites = models.ManyToManyField(Dog, related_name='favourited_by')
```

This can then be retrieved as such `userprofile.favourites.all`